docs/smart-contracts/language-features.md

```diff
@@ -0,0 +1,618 @@
+---
+title: Language features and conventions
+id: lang-features
+sidebar_position: 3
+description: "Learn the basics of Pact programming language features and coding conventions with an introduction to built-in functions and writing your own simple functions."
+---
+
+# Language features and conventions
+
+Pact has a lot in common with other programming languages, but some of its conventions and intentional restrictions are either unique or unlike other programming languages you might be familiar with.
+This part of the documentation presents an overview of Pact language features and conventions you should be familiar with as you start reading and writing smart contract code.
+
+## Command-line interpreter
+
+n the [Quick start](/quickstart), you got a first look at writing Pact smart contract code and using the Pact command-line interpreter, also referred to as the Pact read-eval-print-loop (REPL) environment.
+The Pact REPL enables you to write and execute Pact code interactively from the command-line. but its real power lies in the ability to execute code in smart contract modules—that is, `.pact` files—or in test modules with the `.repl` file extension for testing your code as you go.
+With test modules, you can take advantage of features that are only available to use `.repl` file in the Pact REPL interpreter.
+In most cases, you can use the features provided by the Pact REPL in combination with an integrated development environment (IDE), like Visual Studio Code, to provide an end-to-end development environment. 
+
+## Parenthesis
+
+Pact uses parentheses to enclose each statement in the code.
+Parentheses enclose all module declarations, all function declarations, and any related logic.
+Often, the code requires nested parenthetical statements to resolve the logic.
+For example, the outermost parentheses in the following code contain the **helloWorld** module:
+
+```pact
+(module helloWorld 'admin-keyset
+  (defun hello (name)
+    (format "Hello {}!" [name]))
+)
+```
+
+Within the module `helloworld` declaration, the next set of parentheses contain a `hello` function declaration that includes an expression—also enclosed by parentheses—that uses the `format` built-in function.
+
+## Comments
```

Comment:
Comments are introduced by `;` followed by something until the `\n`. I.e., there is no difference in `;; comment` vs `; comment`.

Reply:
I found this convention called out in some Lisp doc and Kadena seems to lack coding conventions, per se. Ideally, we would adopt some standardization.